lib/date_time/date_time.c

```diff
@@ -731,4 +164,16 @@ int date_time_timestamp_clear(int64_t *unix_timestamp)
 	return 0;
 }
 
-SYS_INIT(date_time_init, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);
+static int date_time_init(const struct device *unused)
```

Comment:
```suggestion
static void date_time_init(const struct device *unused)
```
I suspect that SYS_INIT() expects this function to return `int` so it might not work changing it to `void`